website/content/api-docs/deployments.mdx

```diff
@@ -29,11 +29,13 @@ The table below shows this endpoint's support for
 - `prefix` `(string: "")`- Specifies a string to filter deployments based on
   an ID prefix. Because the value is decoded to bytes, the prefix must have an
   even number of hexadecimal characters (0-9a-f) .This is specified as a query
-  string parameter.
+  string parameter and is backed by a data store index, so it can be used to
+  reduce the number of entries processed by `filter`.
 
 - `namespace` `(string: "default")` - Specifies the target
   namespace. Specifying `*` will return all evaluations across all
-  authorized namespaces.
+  authorized namespaces. This parameter is backed by a data store index, so it
+  can be used to reduce the number of entries processed by `filter`.
```

Comment:
I'm not sure if these additions are good. Feel free to suggest alternatives 😅

Reply:
What if here we focus on the gotcha and link to general filter documentation for more information?

> This parameter is used before any `filter` is applied. See the the [`filter`](https://www.nomadproject.io/docs/api-docs#filter) docs for more information.

Then we could add `-filter` related documentation here: https://www.nomadproject.io/docs/commands#filter This would be the ideal place to call out how the `NOMAD_NAMESPACE` env var will take precedence over any namespace filtering. Although I suppose `api` package users will miss that... :thinking: 

And `?filter=...` related documentation here: https://www.nomadproject.io/api-docs#filter

With perhaps more expansive docs on the query language here? https://www.nomadproject.io/docs/internals/filtering

Or perhaps skip the command and api docs and link directly to an internals doc?

Point being while I think noting these performance impacts is important, I think we also need to guide users toward proper usage of env vars vs query params vs filters.

Reply:
Yeah, I'm planning on having more comprehensive docs about filtering and pagination in a follow-up PR. Basically copy most of [Consul's docs](https://www.consul.io/api-docs/features/filtering) 😅 

Having a CLI section also sounds like a good idea. 

In this PR I will add just the first sentence, and then add a link to the new sections in the follow-up PR.